backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/AsyncAggregatingSubscriber.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.s3;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.Phaser;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Aggregates the results of futures published by a {@link
+ * org.reactivestreams.Publisher<CompletableFuture>}.
+ *
+ * <p>Once the subscription is started, multiple futures are requested from the publisher. New
+ * futures are requested whenever a future completes. The maximum number of requested futures must
+ * be provided in {@link AsyncAggregatingSubscriber#AsyncAggregatingSubscriber(long parallelism)}}
+ *
+ * <p>Futures that complete exceptionally are omitted from the result!
```

Comment:
I think that depends: If **all** S3 calls fail, the future would complete exceptionally because the `listObjectsV2Paginator` api call would fail (at least that's what I think, I'll add a test to verify). If `listObjectsV2Paginator` succeeds but reading all of the manifest object fails the result would be empty.

Reply:
tested: if s3 is completely unavailable the future will complete successfully with an empty result. I'll document this in `S3BackupStore#list`.

Reply:
hmm.. We could improve it later to returns an error/warning to the user, if the list is empty and s3 is unavailable.